actions/actions.py

```diff
@@ -43,6 +44,31 @@
                  }
 
 
+def _check_path_traversal(path_, parent_dir):
+    """Check that 'path_' does not lie outside of the 'parent_dir'.
+
+    This function takes into account possible '../' in 'path_' and also
+    any symlinks that could point somewhere outside the expected 'parent_dir'
+
+    NOTE(mkalcok): This implementation could be improved by using
+                   'os.path.commonpath()'. However it's available only in
+                    py35+.
+
+    :param path_: Path to be tested
+    :param parent_dir: Directory in which the 'path_' must lie
+    :raises: RuntimeError if 'path_' is outside of the 'parent_dir'
+    """
+    full_path = os.path.realpath(path_)
+    if not parent_dir.endswith('/'):
+        parent_dir += '/'
+
+    if os.path.commonprefix([parent_dir, full_path]) != parent_dir:
+        err_msg = "Path traversal detected. '{}' tries to travers out " \
+                  "of {}".format(full_path, parent_dir)
+        log(err_msg, hookenv.CRITICAL)
```

Comment:
I think `CRITICAL` should be reserved for system problems that require intervention and even unit shutdown. All errors that prevent an action to be processed (and possibly resolve by fixing the input) should be logged as `ERROR`.

Reply:
Ok...I'll decrease the level. I just thought that since this is a potential security issue, it might be worth the `CRITICAL`.